assets/src/data/eventespresso/core/actions/remove-relations.js

```diff
@@ -0,0 +1,183 @@
+	return {
+		type: types.REMOVE_RELATED_ENTITIES_FOR_ENTITY,
+		modelName,
+		entityId,
+	};
+}
+
+/**
+ * Action creator for removing relations from the given model from the relations
+ * state.
+ *
+ * @param {string} modelName
+ * @param {number} entityId
+ * @param {string} relationName
+ * @param {Array} relatedEntityIds
+ * @return {
+ * 	{
+ * 		type: string,
+ * 		modelName: string,
+ * 		entityId: number,
+ * 		relationName: string,
+ * 		relationEntityIds: Array
+ * 	}
+ * } An action object.
+ */
+function removeRelatedEntities(
+	modelName,
+	entityId,
+	relationName,
+	relatedEntityIds,
+) {
+	return {
+		type: types.REMOVE_RELATED_ENTITY_IDS,
+		modelName,
+		entityId,
+		relationName,
+		relatedEntityIds,
+	};
+}
+
+/**
+ * Action creator for removing the dirty relation from the state for the given
+ * relation action type (delete or add).
+ *
+ * @param {string} relationName
+ * @param {number} relationEntityId
+ * @param {string} modelName
+ * @param {number} entityId
+ * @param {boolean} addRelation
+ * @return {
+ * 	{
+ * 		relationName: string,
+ * 		relationEntityId: number,
+ * 		modelName: string,
+ * 		entityId: number,
+ * 		type: string,
+ * 		queueType: string
+ * 	}
+ * } An action object
+ */
+function removeDirtyRelationForType(
+	relationName,
+	relationEntityId,
+	modelName,
+	entityId,
+	addRelation = true
+) {
+	const type = addRelation ? types.REMOVE_DIRTY_RELATION_ADDITION :
+		types.REMOVE_DIRTY_RELATION_DELETION;
+	return {
+		relationName,
+		relationEntityId,
+		modelName,
+		entityId,
+		type,
+		queueType: addRelation ?
+			TYPE_QUEUE_RELATION_ADD :
+			TYPE_QUEUE_RELATION_DELETE,
+	};
+}
+
+/**
+ * Action creator for removing the dirty relation from the add relation state
+ * record.
+ *
+ * @param {string} relationName
+ * @param {number} relationEntityId
+ * @param {string} modelName
+ * @param {number} entityId
+ * @return {
+ * 	{
+ * 		relationName: string,
+ * 		relationEntityId: number,
+ * 		modelName: string,
+ * 		entityId: number,
+ * 		type: string,
+ * 		queueType: string
+ * 	}
+ * } An action object.
+ */
+function removeDirtyRelationAddition(
+	relationName,
+	relationEntityId,
+	modelName,
+	entityId,
+) {
+	return removeDirtyRelationForType(
+		relationName,
+		relationEntityId,
+		modelName,
+		entityId,
+	);
+}
+
+/**
+ * An action creator for removing the dirty relation from the delete relation
+ * state record.
+ *
+ * @param {string} relationName
+ * @param {number} relationEntityId
+ * @param {string} modelName
+ * @param {number} entityId
+ * @return {
+ * 	{
+ * 		relationName: string,
+ * 		relationEntityId: number,
+ * 		modelName: string,
+ * 		entityId: number,
+ * 		type: string,
+ * 		queueType: string
+ * 	}
+ * } An action object
+ */
+function removeDirtyRelationDeletion(
+	relationName,
+	relationEntityId,
+	modelName,
+	entityId,
+) {
+	return removeDirtyRelationForType(
+		relationName,
+		relationEntityId,
+		modelName,
+		entityId,
+	);
```

Comment:
whoah, yup good catch.  I'll fix the tests too.

Reply:
Actually the tests are fine.  The tests here are for the dynamically created mode specific action creators, so they are simply testing that the expected function was created and it resolves to the expected action for the model.  It does not actually test that the action shape is as expected.  I didn't write any unit-tests for the non generator actions because they simply return an object so its a bit overkill.

This is an example where having e2e tests would be useful because e2e _would_ usually catch things like "hey this relation should have been removed from state, it wasn't".  I'm surprised this actually didn't surface in my monkey testing.